src/codeu/chat/DerbyStore.java

```diff
@@ -356,6 +367,45 @@ public Message getMessage(Uuid messageid) throws SQLException, IOException {
 		return allMessages;
 	}
 	
+	@Override
+	public int getMessageCount(Uuid userid) throws SQLException {
+		PreparedStatement getMessageCount = conn.prepareStatement(getMessageCountById);
+		getMessageCount.setString(1, removeCharsInUuid(userid.toString()));
+		
+		ResultSet messageCount = getMessageCount.executeQuery();
+		
+		if (messageCount.next()) {
+			return messageCount.getInt(DBColumns.UserMessageCountTable.messageCount);
+		}
+		
+	return 0;
+	}
+
+	@Override
+	public int setUserMessageCount(Uuid userid) throws SQLException {
```

Comment:
Another thing to think about: is it possible for the number of messages to get out of sync with the message table? Basically you have the same data in two different places. This might be okay, but you should make sure it's clear that this function should always be called whenever a user sends a message. Comments comments comments.